components/transfer/transfer-list.component.ts

```diff
@@ -44,7 +44,7 @@ import { NzTransferSearchComponent } from './transfer-search.component';
           [nzChecked]="stat.checkAll"
           (nzCheckedChange)="onItemSelectAll($event)"
           [nzIndeterminate]="stat.checkHalf"
-          [nzDisabled]="stat.shownCount === 0 || disabled"
+          [nzDisabled]="stat.availableCount === 0 || disabled"
```

Comment:
This will change the original action that if all the items are filtered then the select-all button is disabled

what about

```ts
stat.shownCount === 0 || stat.availableCount === 0 || disabled
```